### ISSUE TITLE
R problems

### ISSUE THREAD

**Issue by omnidan** [sirblackheart/elektrasurvey#88]
- [ ] standardize answer ids to make writing R functions that loop through all answers possible
- [ ] standardize question ids (I think we should keep the names as they are right now, e.g. `clusterconf`, not `q5`, but we should still check if they are consistent)
- [ ] maybe we can reconfigure the levels to make working with the data easier - right now the grid questions are fine (they have level 5 with the various answers), but the other ones are level 2 with "Yes" and "Not selected" (right now I'm counting the amount of "Yes" in R)
- [ ] the smiley answers aren't displayed properly in R (all level names are `""`)
- [ ] the immense amount of NAs makes correlations hard (most of the time, either one or the other question is NA, so when correlating two answers, we have to omit NAs and end up with 2-3 answers out of 50)
- [ ] the NAs also make it hard to do cluster analysis (see https://github.com/sirblackheart/elektrasurvey/issues/71#issuecomment-224944536)

**Comment by Namoshek** [comment]
I like the suggestions, but what do you mean with
> maybe we can reconfigure the levels to make working with the data easier - right now the grid questions are fine (they have level 5 with the various answers), but the other ones are level 2 with "Yes" and "Not selected" (right now I'm counting the amount of "Yes" in R)

?

About the smiley answers: Did you export with question/answer tags only, with full answer or combination of them? Tag should work.

**Comment by omnidan** [comment]
@Namoshek In the R export, limesurvey assigns levels to each answer (e.g. grid questions have 5 levels "have not used", "have some experience", etc). With the multiple choice questions, there are just 2 levels ("Yes" and "Not selected"), in some cases it would be easier to deal with just the amount of `Yes`, but it's easy to postprocess this when all the ids are consistent.

I exported with question tags/ids and answer texts (because it's easier to deal with in R, the levels already have the correct names, you don't have to look up the ids, the texts are consistent everywhere, etc...). I think the issue with the smiley answers should be fixed in the new survey, because they also have text in addition to the image.

**Comment by markus2330** [comment]
On Tuesday (before sending it out) we can start filling out the survey ourselves and check if all data is conveniently usable for further processing.

**Comment by Namoshek** [comment]
I'm wondering why there are "Not selected" values for a question as well as N/As... isn't that the same?

**Comment by markus2330** [comment]
It might be the same, but isn't necessarily. For "Not selected" we do not know if it was not selected on purpose or the user simply missed answering the question.

We have (hopefully) constructed the questions in a way that everyone should click at least one answer, so we can transform "Not selected" to NA if nothing was clicked for a particular question.

**Comment by Namoshek** [comment]
You are right, it is hard to tell apart, but NAs seem to be created initially for "in-complete answers", which can be replaced by "Not selected" in case the user answers the question. I guess a question does only count as "answered" (meaning NA will be replaced) if at least one answer is selected. But that doesn't really change the difficulty we face with all these NAs, especially because correlation is only possible if values for two questions are available in the same answer row, which is currently quite rare.

**Comment by markus2330** [comment]
If we get enough answers we can remove all incomplete filled out questionnaires which should simplify it a lot. Maybe we can do it for this pilot manually?

**Comment by Namoshek** [comment]
This is a lot harder than it sounds, because there is not even one answer without any NA yet (e.g. not 5 projects filled out already results in NAs). I'll think about how to handle the issue and maybe I'll create some fake data for testing, but I doubt that we will have a lot of success finding correlations across questions and answers with small data (also because our value range is normally only YES/NOT-SELECTED, which is even weaker than a YES/NO)...

**Comment by markus2330** [comment]
Obviously we cannot find any statistical relevant correlation with the data we have. So for our tests we could duplicate the same answers hundret times and check if the correlation would be found.

For the projects: I think for statistical correclations we need a preprocessing step. We need to merge all projects together to two values:

1. number of projects
2. total year of participations in projects
